homeassistant/components/sensor/__init__.py

```diff
@@ -302,36 +302,62 @@ def state(self) -> Any:
             # Anyways, lets validate the date at least..
             try:
                 value = ciso8601.parse_datetime(str(value))
-            except (ValueError, IndexError) as error:
-                raise ValueError(
-                    f"Invalid date/datetime: {self.entity_id} provide state '{value}', "
-                    f"while it has device class '{device_class}'"
-                ) from error
+            except (ValueError, IndexError):
+                _LOGGER.error(
+                    "Invalid date/datetime: %s provide state '%s', "
+                    "while it has device class '%s'",
+                    self.entity_id,
+                    value,
+                    device_class,
+                )
+                return None
 
             # Convert the date object to a standardized state string.
             if device_class == DEVICE_CLASS_DATE:
                 return value.date().isoformat()
-            return value.isoformat(timespec="seconds")
+
+            if value.tzinfo != timezone.utc:
+                value = value.astimezone(timezone.utc)
+
+            return value.astimezone(timezone.utc).isoformat(timespec="seconds")
 
         # Received a datetime
         if value is not None and device_class == DEVICE_CLASS_TIMESTAMP:
             try:
-                return value.isoformat(timespec="seconds")  # type: ignore
-            except (AttributeError, TypeError) as err:
-                raise ValueError(
-                    f"Invalid datetime: {self.entity_id} has a timestamp device class"
-                    f"but does not provide a datetime state but {type(value)}"
-                ) from err
+                # We cast the value, to avoid using isinstance, but satisfy
+                # typechecking. The error are guarded in this try.
+                value = cast(datetime, value)
+                if value.tzinfo is None:
+                    raise ValueError(
+                        f"Invalid datetime: {self.entity_id} provides state '{value}', "
+                        "which is missing timezone information"
+                    )
+
+                if value.tzinfo != timezone.utc:
+                    value = value.astimezone(timezone.utc)
+
+                return value.isoformat(timespec="seconds")
```

Comment:
Is the check needed, or is this enough?
```suggestion
                return value.isoformat(timespec="seconds")
```

Reply:
uh?

Reply:
Oops, I though it was the same as the previous 🤦